datafusion/spark/src/function/math/trigonometry.rs

```diff
@@ -0,0 +1,193 @@
+        } else {
+            Err(unsupported_data_type_exec_err(
+                CSC_FUNCTION_NAME,
+                "Numeric Type",
+                &arg_types[0],
+            ))
+        }
+    }
+}
+
+fn spark_csc(args: &[ColumnarValue]) -> Result<ColumnarValue> {
+    if args.len() != 1 {
+        return Err(invalid_arg_count_exec_err(
+            CSC_FUNCTION_NAME,
+            (1, 1),
+            args.len(),
+        ));
+    }
+    match &args[0] {
+        ColumnarValue::Scalar(ScalarValue::Float64(value)) => Ok(ColumnarValue::Scalar(
+            ScalarValue::Float64(value.map(|x| 1.0 / x.sin())),
+        )),
+        ColumnarValue::Array(array) => match array.data_type() {
+            DataType::Float64 => Ok(ColumnarValue::Array(Arc::new(
+                array
+                    .as_primitive::<Float64Type>()
+                    .unary::<_, Float64Type>(|x| 1.0 / x.sin()),
+            ) as ArrayRef)),
+            other => Err(unsupported_data_type_exec_err(
+                CSC_FUNCTION_NAME,
+                format!("{}", DataType::Float64).as_str(),
+                other,
+            )),
+        },
+        other => Err(unsupported_data_type_exec_err(
+            CSC_FUNCTION_NAME,
+            format!("{}", DataType::Float64).as_str(),
+            &other.data_type(),
+        )),
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::function::math::trigonometry::{spark_csc, SparkCsc};
+    use crate::function::utils::test::test_scalar_function;
+    use arrow::array::{Array, Float64Array};
+    use arrow::datatypes::DataType::Float64;
+    use datafusion_common::ScalarValue;
+    use datafusion_expr::{ColumnarValue, ScalarUDFImpl};
+    use std::f64::consts::PI;
+    use std::sync::Arc;
+
+    macro_rules! test_trig_float64_invoke {
+        ($FUNC: expr, $INPUT:expr, $EXPECTED:expr) => {
+            test_scalar_function!(
+                $FUNC,
+                vec![ColumnarValue::Scalar(ScalarValue::Float64($INPUT))],
+                $EXPECTED,
+                f64,
+                Float64,
+                Float64Array
+            );
+        };
+    }
+
+    #[test]
+    fn test_csc_invoke() {
+        test_trig_float64_invoke!(SparkCsc::new(), Some(0f64), Ok(Some(f64::INFINITY)));
+    }
+
+    #[test]
+    fn test_csc_array() {
+        let input = Float64Array::from(vec![1f64, 0f64, -1f64]);
+        let expected = Float64Array::from(vec![
+            1.1883951057781212,
+            f64::INFINITY,
+            -1.1883951057781212,
+        ]);
+        let args = ColumnarValue::Array(Arc::new(input));
+
+        if let Ok(ColumnarValue::Array(result_array)) = spark_csc(&[args]) {
+            let output = result_array
+                .as_any()
+                .downcast_ref::<Float64Array>()
+                .unwrap();
+            assert_eq!(output, &expected);
+        } else {
+            panic!("Expected array result");
+        }
+    }
+
+    #[test]
+    fn test_csc_scalar() {
```

Comment:
Are these tests already covered by the SLT tests? We should prefer SLTs where possible

Reply:
Yes, we prefer adding tests to SLT and only fallback to Rust unit tests if it is difficult/impossible to test via SLT